File: react-native/src/service/data/theta-state2.ts
import type { GpsInfo } from './gps-info';

/**
 * Mutable values representing Theta status.
 */
export interface ThetaState2 {
  /**
   * gpsInfo set by API.
   */
  externalGpsInfo: {
    gpsInfo: GpsInfo
  };

  /**
   * gpsInfo from the built-in GPS module.
   */
  internalGpsInfo: {
    gpsInfo: GpsInfo
  };
};

File: kotlin-multiplatform/src/commonMain/kotlin/com/ricoh360/thetableclient/service/data/ThetaState2.kt
package com.ricoh360.thetableclient.service.data

import com.ricoh360.thetableclient.transferred.CameraState2

/**
 * Mutable values representing THETA state
 */
data class ThetaState2(
    /**
     * gpsInfo set by API.
     */
    val externalGpsInfo: StateGpsInfo?,

    /**
     * gpsInfo from the built-in GPS module.
     */
    val internalGpsInfo: StateGpsInfo?,
) {
    internal constructor(value: CameraState2) : this(
        externalGpsInfo = value.externalGpsInfo?.let { StateGpsInfo(value.externalGpsInfo) },
        internalGpsInfo = value.internalGpsInfo?.let { StateGpsInfo(value.internalGpsInfo) },
    )
}
